### PR TITLE
chore/refactor tests

### PR DESCRIPTION
**Pull request by mavogel**
## Steps
- [x] split out test configurations into separate files
- [x] make test image more lightweight
- [x] check resources have the correct entries (like in `TestAccDockerContainer_customized` to avoid bugs like #202 in the future)
- [x] remove `TODOs` in tests and refactor them

## Additional 
- added `insecure_skip_verify` for image pulling on docker image
- added a debug log for each read function in resources
- added check for properties correctly set on the created resource (check for presence in the state file was already there)

### CODE REVIEW

File: internal/provider/data_source_docker_network_test.go [hunk 1]
```diff
@@ -16,7 +16,10 @@ func TestAccDockerNetworkDataSource_basic(t *testing.T) {
 		ProviderFactories: providerFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDockerNetworkDataSourceConfig,
+				Config: `
```
Comment (suzuki-shunsuke) [inline internal/provider/data_source_docker_network_test.go:19]: How about moving this to testdata?

File: internal/provider/data_source_docker_plugin_test.go [hunk 2]
```diff
@@ -23,17 +23,14 @@ func TestAccDockerPluginDataSource_basic(t *testing.T) {
 		ProviderFactories: providerFactories,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDockerPluginDataSourceTest,
+				Config: `
```
Comment (suzuki-shunsuke) [inline internal/provider/data_source_docker_plugin_test.go:26]: How about moving this to testdata?

File: internal/provider/provider_test.go [hunk 3]
```diff
@@ -87,7 +87,8 @@ provider "docker" {
 	}
 }
 data "docker_registry_image" "foobar" {
-	provider = "docker.private"
-	name = "localhost:15000/helloworld:1.0"
+	provider 		     = "docker.private"
+	name     		     = "localhost:15000/helloworld:1.0"
+	insecure_skip_verify = true
```
Comment (suzuki-shunsuke) [inline internal/provider/provider_test.go:92]:
```suggestion
	provider             = "docker.private"
	name                 = "localhost:15000/helloworld:1.0"
	insecure_skip_verify = true

```

File: internal/provider/resource_docker_config_test.go [hunk 4]
```diff
@@ -5,12 +5,31 @@ import (
...
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
 func TestAccDockerConfig_basic(t *testing.T) {
 	ctx := context.Background()
+	var c swarm.Config
+
+	testCheckConfigInspect := func(*terraform.State) error {
+		if c.Spec.Name == "" {
+			return fmt.Errorf("Config Spec.Name is wrong: %v", c.Spec.Name)
```
Comment (suzuki-shunsuke) [inline internal/provider/resource_docker_config_test.go:19]:
```suggestion
			return errors.New("Config Spec.Name is empty")
```

File: internal/provider/resource_docker_config_test.go [hunk 5]
```diff
@@ -5,12 +5,31 @@ import (
...
+		}
+
+		if len(c.Spec.Data) == 0 {
+			return fmt.Errorf("Config Spec.Data is wrong: %v", c.Spec.Data)
```
Comment (suzuki-shunsuke) [inline internal/provider/resource_docker_config_test.go:23]:
```suggestion
			return errors.New("Config Spec.Data is empty")
```

File: internal/provider/data_source_docker_network_test.go (hunk 1 above)
Comment (mavogel) [reply internal/provider/data_source_docker_network_test.go:19]: done

File: internal/provider/data_source_docker_plugin_test.go (hunk 2 above)
Comment (mavogel) [reply internal/provider/data_source_docker_plugin_test.go:26]: done

File: internal/provider/provider_test.go (hunk 3 above)
Comment (mavogel) [reply internal/provider/provider_test.go:92]: done

File: internal/provider/resource_docker_config_test.go (hunk 4 above)
Comment (mavogel) [reply internal/provider/resource_docker_config_test.go:19]: done

File: internal/provider/resource_docker_config_test.go (hunk 5 above)
Comment (mavogel) [reply internal/provider/resource_docker_config_test.go:23]: done

**Review by suzuki-shunsuke — APPROVED** [review]

**Pull request merged**